docs/pages/api/access-plugin.mdx

```diff
@@ -0,0 +1,738 @@
+this function below `createRow`:
+
+```go
+func (g *googleSheetsPlugin) updateRow(ar types.AccessRequest, rowNum int64) error {
+
+	row := g.makeRowData(ar)
+
+	req := sheets.BatchUpdateSpreadsheetRequest{
+		Requests: []*sheets.Request{
+			{
+				UpdateCells: &sheets.UpdateCellsRequest{
+
+					Fields: "*",
+					Start: &sheets.GridCoordinate{
+						RowIndex: rowNum,
+					},
+					Rows: []*sheets.RowData{
+						row,
+					},
+				},
+			},
+		},
+	}
+	resp, err := g.sheetsClient.BatchUpdate(spreadSheetID, &req).Do()
+
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	if resp.HTTPStatusCode == 201 || resp.HTTPStatusCode == 200 {
+		fmt.Println("Successfully updated a row")
+	} else {
+		fmt.Printf(
+			"Unexpected response code updating a row: %v\n",
+			resp.HTTPStatusCode,
+		)
+	}
+
+	return nil
+
+}
+```
+
+The only difference between `updateRow` and `createRow` is that we send a
+`&sheets.UpdateCellsRequest` instead of a `&sheets.AppendCellsRequest`. This
+function takes the number of a row within the spreadsheet to update and sends a
+request to update that row with information from the provided Access Request.
+
+### Determine where to update the spreadsheet
+
+Below `updateRow`, add this function:
+
+```go
+func (g *googleSheetsPlugin) updateSpreadsheet(ar types.AccessRequest) error {
+
+	s, err := g.sheetsClient.Get(spreadSheetID).IncludeGridData(true).Do()
+
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	if len(s.Sheets) != 1 {
+		return errors.New("the spreadsheet must have a single sheet")
+	}
+
+	for _, d := range s.Sheets[0].Data {
+		for i, r := range d.RowData {
+			if r.Values[0] != nil &&
+				r.Values[0].UserEnteredValue != nil &&
+				r.Values[0].UserEnteredValue.StringValue != nil &&
+				*r.Values[0].UserEnteredValue.StringValue == ar.GetName() {
+				if err := g.updateRow(ar, int64(i)); err != nil {
+					return err
+				}
+				fmt.Println("Updated a spreadsheet row.")
+			}
+		}
+	}
+	return nil
+}
+```
+
+`updateSpreadSheet` takes a `types.AccessRequest`, gets the latest data from
+your spreadsheet, determines which row to update, and calls `updateRow`
+accordingly. It uses linear search to look up the first column within each row
+of the sheet and check whether that column matches the ID of the Access Request.
+It then calls `updateRow` with the Access Request and the row's number.
+
+### Handle incoming Access Requests
+
+Next, we will determine whether to call `updateSpreadsheet` or `createRow`
+depending on whether an incoming Access Request event is a new Access Request or
+an update to an existing one. Add the following function below
+`updateSpreadsheet`:
+
+```go
+func (g *googleSheetsPlugin) handleEvent(ctx context.Context, event types.Event) error {
+
+	if event.Resource == nil {
+		return nil
+	}
+
+	r := event.Resource.(types.AccessRequest)
+
+	// Ignore Access Requests that are not Role Requests
+	if len(r.GetRoles()) == 0 {
```

Comment:
Don't resource access requests have role(s) too?

Reply:
This was a bad assumption on my part based on not being familiar enough with Resource Access Requests. I'll remove this.